src/python/era_5g_netapp_client/era_5g_netapp_client/client.py

```diff
@@ -0,0 +1,282 @@
+            use_middleware (bool, optional): Defines if the NetApp should be deployed by middleware. 
+                If False, the netapp_uri and netapp_port need to be defined, otherwise, 
+                they need to be left None. Defaults to True.
+            netapp_uri (str, optional): The URI of the NetApp interface. Defaults to None.
+            netapp_port (int, optional): The port of the NetApp interface. Defaults to None.
+        
+        Raises:
+            FailedToConnect: When connection to the middleware could not be setupped or 
+                login failed
+            FailedToObtainPlan: When the plan was not sucessfully returned from
+                the middleware
+        """
+
+        # temporary removed asserts until the middleware will provide neccessary info
+
+        # if middleware is used, the netapp_uri and port should be obtained from it
+        #assert use_middleware and not netapp_uri and not netapp_port
+        # if middleware is not used, the netapp_uri and port needs to be provided
+        #assert not use_middleware and netapp_uri and netapp_port
+
+        self.num_steps = 0
+        self.action_seq_ids = []
+
+
+        self.sio = socketio.Client()
+        self.s = requests.session()
+        self.host = host.rstrip('/')
+        self.resource_lock = resource_lock
+        self.netapp_host = netapp_uri
+        self.netapp_port = netapp_port
+        self.use_middleware = use_middleware
+        self.sio.on("message", results_event, namespace='/results')
+        self.sio.on("connect", self.on_connect_event, namespace='/results')
+        self.sesion_cookie = None
+        self.plan = None
+        self.action_plan_id = None
+        self.resource_checker = None
+        self.user_id = user_id
+        self.password = password
+        self.task_id = task_id
+        self.token = None
+        self.wait_for_netapp = wait_for_netapp
+        if self.use_middleware:
+            self.connect_to_middleware()
+
+        
+    def connect_to_middleware(self):
+        try:            
+            # connect to the middleware
+            self.token = self.gateway_login(self.user_id, self.password)
+
+            self.plan = self.gateway_get_plan(self.task_id, self.resource_lock)  # Get the plan by sending the token and TaskId
+
+            self.resource_checker = MiddlewareResourceChecker(logger, "resource_checker", self.token, self.action_plan_id, self.build_middleware_api_endpoint("orchestrate/orchestrate/plan"))
+            self.resource_checker.start(daemon=True)
+            if self.use_middleware and self.wait_for_netapp:
+                self.wait_until_netapp_ready()
+                print(self.resource_checker.is_ready())
+                self.load_netapp_uri()
+            print('Feedback: Got new plan successfully.')
+        except Exception as ex:
+            self.deleteAllResources()
+            raise ex
+
+    
+        
+    def register(self, args=dict()) -> str:
```

Comment:
@Kapim Check this https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
Wasn't aware of this, thank you, changed to None instead.